internal/cri/server/stats_test.go

```diff
@@ -0,0 +1,259 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package server
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/containerd/containerd/v2/api/types"
+	containerstore "github.com/containerd/containerd/v2/internal/cri/store/container"
+	sandboxstore "github.com/containerd/containerd/v2/internal/cri/store/sandbox"
+	"github.com/containerd/containerd/v2/internal/cri/store/stats"
+	"github.com/containerd/containerd/v2/protobuf"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetUsageNanoCores(t *testing.T) {
+	timestamp := time.Now()
+	secondAfterTimeStamp := timestamp.Add(time.Second)
```

Comment:
Should we add another test for microseconds or emulate parallel requests to guarantee the old behavior do not appear as a regression?

Reply:
I can add a test where it is one milli second afterward to make sure the calculation would make that different value. 

I thought I added a test where it made sure the cached value was returned, but maybe I missed it...